models/demos/falcon7b/tests/test_perplexity_falcon.py

```diff
@@ -0,0 +1,207 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import torch
+from loguru import logger
+from transformers import AutoTokenizer
+from tqdm import tqdm
+import time
+from models.demos.falcon7b.tt.falcon_causallm import TtFalconCausalLM
+from models.demos.falcon7b.tt.model_config import get_model_config
+from models.demos.falcon7b.tests.test_utils import initialize_kv_cache, load_hf_model
+from models.datasets.llm_dataset_utils import prepare_textgen_dataset, prepare_textgen_dataloader
+from models.utility_functions import is_wormhole_b0, get_devices_for_t3000, tt_tensors_to_torch_tensors
+
+
+def eval_perplexity(tt_FalconCausalLM, dataloader, llm_mode, batch_size, seq_len, kv_cache, configuration):
```

Comment:
I'd rather call function `calculate_perplexity` instead of `eval_perplexity` since it's primary role is to return perplexity and negative log likelihood, right?

Reply:
Agreed, updated the name